From 0f8d9923eef3b10015b6fb102e0361350eac7123 Mon Sep 17 00:00:00 2001
From: Patrick Schwaiger <39056009+Patrick112233@users.noreply.github.com>
Date: Mon, 29 Dec 2025 12:16:41 +0100
Subject: [PATCH] fix dyn linker macro for cross compile

---
 m4/dyn_linker.m4 | 14 +++++++++-----
 1 file changed, 9 insertions(+), 5 deletions(-)

diff --git a/m4/dyn_linker.m4 b/m4/dyn_linker.m4
index c0c8aa3..45d589f 100644
--- a/m4/dyn_linker.m4
+++ b/m4/dyn_linker.m4
@@ -1,11 +1,15 @@
 AC_DEFUN([LD_SO_PATH],
 [
-  bash_path=`command -v bash`
-  xpath1=`readelf -e $bash_path | grep Requesting | sed 's/.$//' | rev | cut -d" " -f1 | rev`
-  xpath=`realpath $xpath1`
-  if test ! -f "$xpath" ; then
-    AC_MSG_ERROR([Cant find the dynamic linker])
+  AC_ARG_WITH([system-ld-so],
+    [AS_HELP_STRING([--with-system-ld-so=PATH],
+      [specify the path to the dynamic linker])],
+    [xpath="$withval"],
+    [AC_MSG_ERROR([--with-system-ld-so=PATH must be specified])])
+  
+  if test -z "$xpath" ; then
+    AC_MSG_ERROR([Dynamic linker path cannot be empty])
   fi
+  
   echo "dynamic linker is.....$xpath"
   AC_DEFINE_UNQUOTED(SYSTEM_LD_SO, ["$xpath"], [dynamic linker])
 ])
